test/span.js

```diff
@@ -185,31 +184,41 @@ describe('span should', () => {
         assert.isOk(unnormalizedKey in Span._getBaggageHeaderCache());
     });
 
-    describe('with deferred sampling', () => {
```

Comment:
your new test does not cover the use case of changing the operation name, which was the main reason for deferred samping logic